build-system/tasks/dep-check.js

```diff
@@ -204,9 +203,12 @@ function getGraph(entryModule) {
  */
 function getEntryModule(extensionFolder) {
   var extension = path.basename(extensionFolder);
-  var versions = extensionsVersions[extension] || ['0.1'];
-  return versions.map(v =>
-      `${extensionFolder}/${v}/${extension}.js`);
+  return fs.readdirAsync(extensionFolder).then(dirItems => {
```

Comment:
Ahhh I think I know what happen here. `getEntryModule` needs to return an Array, but with `fs.readdirAsync` it is returning a promise and no dep-check is running for any extension. 
I am able to see `dep-check` failure after switching to `fs.readdirSync`